### PR TITLE
multiple glob patterns

### PR DESCRIPTION
**Pull request by ericsciple**
This change enables multiple patterns

### CODE REVIEW

File: packages/glob/src/internal-pattern-helper.ts [hunk 1]
```diff
@@ -105,35 +73,35 @@ export function match(patterns: Pattern[], itemPath: string): MatchKind {
...
-        pattern.segments[pattern.segments.length - 1] !== '**')
-    ) {
-      // Push
-      result.push(new Pattern(pattern.negate, pattern.segments.concat('**')))
-    }
-  }
-
-  return result
-}
+// /**
```
Comment (ericsciple) [inline packages/glob/src/internal-pattern-helper.ts]: remove

File: packages/glob/src/internal-pattern-helper.ts [hunk 2]
```diff
@@ -1,41 +1,9 @@
-import * as core from '@actions/core'
 import * as pathHelper from './internal-path-helper'
-import {IGlobOptions} from './internal-glob-options'
 import {MatchKind} from './internal-match-kind'
 import {Pattern} from './internal-pattern'
 
 const IS_WINDOWS = process.platform === 'win32'
 
-/**
```
Comment (ericsciple) [inline packages/glob/src/internal-pattern-helper.ts]: this moved

File: packages/glob/src/internal-globber.ts [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+      else if (match & MatchKind.File) {
+        yield item.path
+      }
+    }
+  }
+
+  /**
+   * Parses the input options and patterns to construct a new GlobberImpl instance.
+   */
+  static async parse(input: string): Promise<DefaultGlobber> {
```
Comment (ericsciple) [inline packages/glob/src/internal-globber.ts]: this parse function is new, everything else in this file basically moved into this class (was in glob.ts)

File: packages/glob/src/internal-globber.ts [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+import {SearchState} from './internal-search-state'
+
+const IS_WINDOWS = process.platform === 'win32'
+
+export {GlobOptions}
+
+/**
+ * Used to match files and directories
+ */
+export interface Globber {
```
Comment (ericsciple) [inline packages/glob/src/internal-globber.ts]: the new interface for getting search paths and globbing for files

File: packages/glob/src/internal-glob-options.ts [hunk 5]
```diff
@@ -1,9 +1,11 @@
...
+ * Options to control globbing behavior
+ */
+export interface GlobOptions {
   /**
-   * Indicates whether to follow symbolic links. Generally should be true
-   * unless deleting files.
+   * Indicates whether to follow symbolic links
    *
-   * @default true
+   * @default false
```
Comment (ericsciple) [inline packages/glob/src/internal-glob-options.ts]: flipped to false

File: packages/glob/src/internal-glob-options-helper.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import * as core from '@actions/core'
+import {GlobOptions} from './internal-glob-options'
+
+/**
+ * Returns a copy with defaults filled in.
+ */
+export function getOptions(copy?: GlobOptions): GlobOptions {
```
Comment (ericsciple) [inline packages/glob/src/internal-glob-options-helper.ts]: this function basically just moved from internal-pattern-helper.ts

also the default for followSymbolicLinks flipped to false

File: packages/glob/src/glob.ts [hunk 7]
```diff
@@ -1,183 +1,27 @@
...
-export async function glob(
-  pattern: string,
-  options?: IGlobOptions
-): Promise<string[]> {
-  const result: string[] = []
-  for await (const itemPath of globGenerator(pattern, options)) {
-    result.push(itemPath)
-  }
-  return result
+export async function getInput(
```
Comment (ericsciple) [inline packages/glob/src/glob.ts]: now this file is only used to export the interfaces (Globber, GlobOptions) and instantiate a globber (getInput, parse)

everything else moved into internal-globber.ts

Comment (ericsciple) [inline packages/glob/src/glob.ts]: i'm going to remove the `getInput` function and just keep `parse`

Comment (ericsciple) [reply packages/glob/src/glob.ts]: i might change it to `create`? suggestions welcome

File: packages/glob/src/glob.ts [hunk 8]
```diff
@@ -1,183 +1,13 @@
...
-import {SearchState} from './internal-search-state'
+import {Globber, DefaultGlobber} from './internal-globber'
+import {GlobOptions} from './internal-glob-options'
 
-export {IGlobOptions}
+export {Globber, GlobOptions}
 
 /**
- * Returns files and directories matching the specified glob pattern.
+ * Constructs a globber
```
Comment (ericsciple) [inline packages/glob/src/glob.ts]: now this file is only used to export the interfaces (`Globber`, `GlobOptions`) and instantiate a globber (`create`)

everything else moved into internal-globber.ts

File: packages/glob/src/internal-glob-options.ts (hunk 5 above)
Comment (ericsciple) [reply packages/glob/src/internal-glob-options.ts]: flip back to true :)

File: packages/glob/src/internal-globber.ts [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+      else if (match & MatchKind.File) {
+        yield item.path
+      }
+    }
+  }
+
+  /**
+   * Constructs a DefaultGlobber
+   */
+  static async create(
```
Comment (ericsciple) [inline packages/glob/src/internal-globber.ts]: this `create` function is new, everything else in this file basically moved into this class (was in glob.ts)

File: packages/glob/src/internal-glob-options-helper.ts (hunk 6 above)
Comment (TingluoHuang) [reply packages/glob/src/internal-glob-options-helper.ts]: true or false?

Comment (ericsciple) [reply packages/glob/src/internal-glob-options-helper.ts]: flipped back to true!

**Review by abirhasan132 — COMMENTED** [review]
#287

**Review by abirhasan132 — APPROVED** [review]
152e3137c2075505311d5ae6fabee1aea672d6b3

**Pull request merged**